R/create_project.R

```diff
@@ -0,0 +1,256 @@
+#' Creates a pre-populated project for DfE R
+#'
+#' @param path Path of the new project
+#' @param init_renv Boolean; initiate renv in the project.
+#' Default is set to true.
+#' @param include_structure_for_pkg Additional folder structure for
+#' package development
+#' @param create_publication_proj Should the folder structure be for a
+#' publication project
+#' @param include_github_gitignore Should a strict .gitignore file for
+#' GitHub be created
+#' @param ... Additional parameters, currently not used
+#'
+#' @details This function creates a new project with a custom folder structure.
+#' It sets up the `R/` folder and template function scripts,
+#' initializes `{testthat}` and adds tests for the function scripts,
+#' builds the core project structure, creates a .gitignore file,
+#' creates a readme, and optionally initializes `{renv}`.
+#'
+#' @return No return values, the project and its contents are created
+#' @importFrom utils installed.packages
+#' @importFrom withr with_dir
+#' @importFrom usethis create_package create_project use_testthat use_test
+#' @importFrom usethis proj_set
+#' @import testthat
+#' @rawNamespace import(renv, except = run)
+#' @import rmarkdown
+#' @import stringr
+#' @export
+#' @examples
+#' \dontrun{
+#' # Load the dfeR package (or even better call it explicitly)
+#' library(dfeR)
+#'
+#' # Define the path for the new project
+#' path <- "/path/to/your/new/project"  # replace with your actual path
+#'
+#' # Call the function to create a new project
+#' dfeR::create_project(
+#'   path = path,
+#'   init_renv = TRUE,
+#'   include_structure_for_pkg = TRUE,
+#'   create_publication_proj = FALSE,
+#'   include_github_gitignore = TRUE
+#' )
+#' }
+create_project <- function(
+    path,
+    init_renv = TRUE,
+    include_structure_for_pkg,
```

Comment:
Yep agreed, will change these. I have set include_structure_for_pkg() and create_publication_proj() to default as FALSE but left include_github_gitignore() as open, as think it would be good practice for the user to decide this?

Will include a `stop()` to check all params are specified.